tests/bluetooth/tester/src/btp_cap.c

```diff
@@ -0,0 +1,1028 @@
+	if (cp->cc_ltvs_len != 0) {
+		codec_cfg.data_len = cp->cc_ltvs_len;
+		memcpy(codec_cfg.data, ltv_ptr, cp->cc_ltvs_len);
+		ltv_ptr += cp->cc_ltvs_len;
+	}
+
+	if (cp->metadata_ltvs_len != 0) {
+		codec_cfg.meta_len = cp->metadata_ltvs_len;
+		memcpy(codec_cfg.meta, ltv_ptr, cp->metadata_ltvs_len);
+	}
+
+	err = cap_unicast_setup_stream(conn, cp->ase_id, cp->cis_id, cp->cig_id, &codec_cfg, &qos);
+	bt_conn_unref(conn);
+	if (err != 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t btp_cap_unicast_audio_start(const void *cmd, uint16_t cmd_len,
+					   void *rsp, uint16_t *rsp_len)
+{
+	int err;
+	size_t stream_count = 0;
+	struct unicast_group *u_group;
+	const struct btp_cap_unicast_audio_start_cmd *cp = cmd;
+	struct bt_cap_unicast_audio_start_param start_param;
+	struct bt_cap_unicast_audio_start_stream_param stream_params[ARRAY_SIZE(set_members) *
+								     MAX_STREAMS_COUNT];
+
+	LOG_DBG("");
+
+	err = tester_bap_client_create_unicast_group(cp->cig_id, &u_group);
+	if (err != 0) {
+		LOG_ERR("Failed to create unicast group");
+
+		return BTP_STATUS_FAILED;
+	}
+
+	for (size_t conn_index = 0; conn_index < (size_t)ARRAY_SIZE(set_members); conn_index++) {
+		struct unicast_connection *u_conn = tester_bap_unicast_conn_get(conn_index);
+
+		if (u_conn->end_points_count == 0) {
+			/* Connection not initialized */
+			continue;
+		}
+
+		for (size_t i = 0; i < ARRAY_SIZE(u_conn->streams); i++) {
+			struct bt_cap_unicast_audio_start_stream_param *stream_param;
+			struct unicast_stream *u_stream = &u_conn->streams[i];
+
+			if (!u_stream->in_use || u_stream->cig_id != cp->cig_id) {
+				continue;
+			}
+
+			stream_param = &stream_params[stream_count++];
+			stream_param->stream = stream_unicast_to_cap(u_stream);
+			stream_param->codec_cfg = &u_stream->codec_cfg;
+			stream_param->member.member = bt_conn_lookup_addr_le(BT_ID_DEFAULT,
+									     &u_conn->address);
+			stream_param->ep = tester_bap_end_point_find(u_conn, u_stream->ase_id);
+		}
+	}
+
+	start_param.type = BT_CAP_SET_TYPE_AD_HOC;
+	start_param.count = stream_count;
+	start_param.stream_params = stream_params;
+
+	err = bt_cap_initiator_unicast_audio_start(&start_param, u_group->cig);
+	if (err != 0) {
+		LOG_ERR("Failed to start unicast audio: %d", err);
+
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t btp_cap_unicast_audio_update(const void *cmd, uint16_t cmd_len,
+					    void *rsp, uint16_t *rsp_len)
+{
+	int err;
+	const uint8_t *data_ptr;
+	const struct btp_cap_unicast_audio_update_cmd *cp = cmd;
+	struct bt_cap_unicast_audio_update_param stream_params[ARRAY_SIZE(set_members) *
+								     MAX_STREAMS_COUNT];
+
+	LOG_DBG("");
+
+	if (cp->stream_count == 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	data_ptr = cp->update_data;
+	for (size_t i = 0; i < cp->stream_count; i++) {
+		struct unicast_connection *u_conn;
+		struct unicast_stream *u_stream;
+		struct bt_conn *conn;
```

Comment:
```suggestion
		const struct bt_conn *conn;
```

Reply:
So far I have avoided `const` near every`struct bt_conn *conn` so as not to have to cast at `bt_conn_unref((struct bt_conn *)conn);`. Do you think this is a bad practice and every similar usage should be corrected?

Reply:
Ah, sorry, yeah, it cannot be `const` if you are `unref`'ing it. My bad :)